### ISSUE TITLE
In-place upgrades for cloudsql resources

### ISSUE THREAD

**Issue by songalbert** [hashicorp/terraform-provider-google#11903]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

Terraform created cloudsql databases should be able to be upgraded in-place without causing issues of duplicate resources or hardcoded database versions. With the [GCP console in-place upgrade feature](https://cloud.google.com/sql/docs/postgres/upgrade-major-db-version-inplace) available, it can cause problems if you are using terraform for your cloudsql maintenance.

### New or Affected Resource(s)

google cloudsql postgres

### Potential Terraform Configuration

```tf
# Propose what you think the configuration to take advantage of this feature should look like.
# We may not use it verbatim, but it's helpful in understanding your intent.
```

### References

* #0000

**Comment by shuyama1** [comment]
@songalbert Sorry for the late investigation on this issue. It seems like we recently implemented the functionality of in-place major version upgrade for sql instance. The feature is released in version 4.34.0. Is this something you are looking for?

**Comment by shuyama1** [comment]
Closing this issue as it may have already implemented. Feel free to reopen this issue or create a new one if you think the feature is not yet supported in the providers.
